### PR TITLE
Support Artifactory repo operations with templates

### PR DESCRIPTION
**Pull request by liron-shalom**
Add the ability of template creation using interactive questionnaire.
Add commands for repo updare and create using templates.
Add cpmmand for repo delete.

### CODE REVIEW

File: artifactory/commands/repository/template.go [hunk 1]
```diff
@@ -0,0 +1,886 @@
...
+	}
+	iq.AnswersMap[ContentSynchronisation] = answer
+	return "", nil
+}
+
+// After an optionla value was chosen we'll ask for its value
+func optionalKeyCallback(iq *utils.InteractiveQuestionnaire, key string) (value string, err error) {
+	if key != utils.WriteAndExist {
+		valueQuestion := iq.QuestionsMap[key]
+		// Since we are using default question in most of the cases we set set the mak key here
```
Comment (eyalbe4) [inline artifactory/commands/repository/template.go]: set set --> set

File: artifactory/commands/utils/interactive_questionnaire.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+func (iq *InteractiveQuestionnaire) AskQuestion(question QuestionInfo) (value string, err error) {
+
+	var answer string
+	if question.Options != nil {
+		answer = AskFromList(question.Msg, question.PromptPrefix, question.AllowVars, question.Options)
+	} else {
+		answer = AskString(question.Msg, question.PromptPrefix)
+	}
+	if question.Writer != nil {
+		//err = question.Writer(&iq.AnswersMap, question.MapKey, answer)
```
Comment (eyalbe4) [inline artifactory/commands/utils/interactive_questionnaire.go]: This comment should be probably removed.

File: artifactory/commands/utils/interactive_questionnaire.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+	Callback     questionCallback
+}
+
+const (
+	PressTabMsg      = " (press Tab for options):"
+	InvalidAnswerMsg = "Invalid answer. Please select value from the suggestions list."
+	VariableUseMsg   = " You may use dynamic variable in the form of ${key}."
+	EmptyValueMsg    = "The value can not be empty. Please enter a valid value:"
+	OptionalKey      = "OptionalKey"
+	WriteAndExist    = ":x"
```
Comment (eyalbe4) [inline artifactory/commands/utils/interactive_questionnaire.go]: I think it should be **WriteAndExit**. I actually suggest renaming it to **SaveAndExit**.

File: docs/artifactory/repocreate/help.go [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package repocreate
+
+const Description = "Create a new repository in Artifactory."
+
+var Usage = []string{`jfrog rt rc <template path>`}
+
+const Arguments string = `	template path
+		Specifies the local file system path for the template file to be used for the repository creation.`
```
Comment (eyalbe4) [inline docs/artifactory/repocreate/help.go]: Let's add to the end of the description:
The template can be created using the "jfrog rt rpt <template path>" command.
Let's add this to the repoupdate help as well.

File: artifactory/cli.go [hunk 5]
```diff
@@ -605,6 +610,58 @@ func GetCommands() []cli.Command {
...
+			UsageText:    repoupdate.Arguments,
+			ArgsUsage:    common.CreateEnvVars(),
+			BashComplete: common.CreateBashCompletionFunc(),
+			Action: func(c *cli.Context) error {
+				return repoCmd(c, true)
+			},
+		},
+		{
+			Name:         "repo-delete",
+			Aliases:      []string{"rd"},
```
Comment (eyalbe4) [inline artifactory/cli.go]: rd --> rdel
(to be consistent with our other delete commands).
Let's change in help.go as well.

File: docs/artifactory/repodelete/help.go [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package repodelete
+
+const Description = "Remove a repository from Artifactory."
```
Comment (eyalbe4) [inline docs/artifactory/repodelete/help.go]: Remove a repository... --> Permanently delete a repository with all of its content...

File: artifactory/commands/utils/interactive_questionnaire.go (hunk 2 above)
Comment (liron-shalom) [reply artifactory/commands/utils/interactive_questionnaire.go]: Actually this is the right line should stay -
We want to keep this questionnaire generic as possible, so we use the writer and not write every answer as string.
I added new writers at template.go that writes every question as a string.

**Pull request merged**